docs/managed-flags.rst

```diff
@@ -15,6 +15,15 @@ directly; any base layer should instead use
 :func:`~charms.reactive.flags.register_trigger` to "wrap" the automatic flag
 with a layer-specific flag that can be safely used within that layer.
 
+It is worth mentioning that any set flag will execute on every hook run as
+long as it stays set. This means that code inside a function wrapped in,
+for example, ``endpoint.{endpoint_name}.joined`` will run every time hooks
```

Comment:
I would say that "all handlers will be evaluated against a set of flags currently set. A new evaluation will happen after every handler execution within the same Juju hook execution".

"<flag_name> will run" sounds a bit confusing.